crates/ruff/src/checkers/ast.rs

```diff
@@ -1810,6 +1814,14 @@ where
                     }
                 }
 
+                if self.settings.rules.enabled(&Rule::GlobalStatement) {
+                    for target in targets.iter() {
+                        if let ExprKind::Name { id, .. } = &target.node {
+                            pylint::rules::global_statement(self, id);
+                        }
+                    }
+                }
```

Comment:
I think, instead, we should put this in the `StmtKind::Global { names } => { ... }` branch around line 341. That is: we should just trigger this rule when we declare the global.

Reply:
If I'm understanding correctly, what you want is for the behavior to be "warn whenever any global variable is declared", which is different from pylint's [original definition of the W0603 rule](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/global-statement.html): 

> Used when you use the "global" statement to update a global variable

Want to confirm if that's ok?

Reply:
Ohh interesting, I totally missed that. Then I think you're on the right track here. The rule is maybe a little strange, since if you're not using `global` to update or otherwise mutate a variable, then the `global` is effectively useless anyway, right? Like why else define a `global`? But I generally prefer to maintain parity with upstream rules unless we have good reasons not to.

Reply:
Haha yeah I agree, the original pylint rule feels a bit weird already, also there are 3 separate warnings that the user can configure but regardless, the range reported in all of them seems to always point to the global statement 😕 

```
[09:01:35 | last: 0s] (  0) | /tmp
igozali@hostname $ bat main.py
───────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: main.py
───────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ def do_something():
   2   │     global a
───────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

[09:01:39 | last: 0s] (  0) | /tmp
igozali@hostname $ pylint -d C --score=n main.py
************* Module main
main.py:2:4: W0602: Using global for 'a' but no assignment is done (global-variable-not-assigned)
```

```
[09:01:54 | last: 3s] (  0) | /tmp
igozali@hostname $ bat main.py
───────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: main.py
───────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ def do_something():
   2   │     global a
   3   │     a = 5
───────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

[09:01:56 | last: 0s] (  0) | /tmp
igozali@hostname $ pylint -d C --score=n main.py
************* Module main
main.py:2:4: W0601: Global variable 'a' undefined at the module level (global-variable-undefined)
```

```
[09:02:20 | last: 3s] (  0) | /tmp
igozali@hostname $ bat main.py
───────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: main.py
───────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ a = 0
   2   │
   3   │ def do_something():
   4   │     global a
   5   │     a = 5
───────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

[09:02:22 | last: 0s] (  0) | /tmp
igozali@hostname $ pylint -d C --score=n main.py
************* Module main
main.py:4:4: W0603: Using the global statement (global-statement)
```